### PR TITLE
feat(server): add Netlify adapter

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Add adapter and guide for publishing Expo Router v3 websites/api routes to Netlify.

### CODE REVIEW

File: docs/pages/router/reference/api-routes.mdx [hunk 1]
```diff
@@ -282,3 +282,85 @@ Start the server with `node` command:
 <Terminal cmd={['$ node server.js']} />
 
 </Step>
+
+### Netlify
+
+> This is experimental and subject to breaking changes. We have no continuous tests against this configuration.
```
Comment (amandeepmittal) [inline docs/pages/router/reference/api-routes.mdx:288]:
```suggestion
> **warning** This is experimental and subject to breaking changes. We have no continuous tests against this configuration.
```

Let's add a warning callout here.

File: docs/pages/router/reference/api-routes.mdx [hunk 2]
```diff
@@ -282,3 +282,85 @@ Start the server with `node` command:
...
+
+<Step label="1">
+
+Create a server entry file, all requests will be delegated through this middleware. The exact file location is important.
```
Comment (amandeepmittal) [inline docs/pages/router/reference/api-routes.mdx:292]:
```suggestion
Create a server entry file. All requests will be delegated through this middleware. The exact file location is important.
```

File: docs/pages/router/reference/api-routes.mdx [hunk 3]
```diff
@@ -282,3 +282,85 @@ Start the server with `node` command:
...
+});
+
+module.exports = { handler };
+```
+
+</Step>
+
+<Step label="2">
+
+Create a Netlify configuration file to redirect all requests to the server function.
```
Comment (amandeepmittal) [inline docs/pages/router/reference/api-routes.mdx:311]:
```suggestion
Create a Netlify configuration file at the root of your project to redirect all requests to the server function.
```

Let's add context as to where this file needs to be created.

File: docs/pages/router/reference/api-routes.mdx [hunk 4]
```diff
@@ -282,3 +282,85 @@ Start the server with `node` command:
...
+
+<Step label="3">
+
+After you have created the configuration files, you can build the website and functions with Expo CLI:
+
+<Terminal cmd={['$ npx expo export -p web']} />
+
+</Step>
+
+<Step label="4">
```
Comment (amandeepmittal) [inline docs/pages/router/reference/api-routes.mdx:345]: In step 4, can we redirect developers to follow the steps here: https://docs.expo.dev/distribution/publishing-websites/#netlify. Or the steps defined in the link are not relevant here?

File: docs/pages/router/reference/api-routes.mdx [hunk 5]
```diff
@@ -282,3 +282,85 @@ Start the server with `node` command:
...
+  ]}
+/>
+
+You can now visit your website at the URL provided by Netlify CLI. Running `netlify deploy --prod` will publish to the production URL.
+
+</Step>
+
+<Step label="5">
+
+If you're using any environment variables or `.env` files, you'll need to add them to Netlify. You can do this by going to the "Site settings" page and adding them to the "Build & deploy" section.
```
Comment (amandeepmittal) [inline docs/pages/router/reference/api-routes.mdx:364]:
```suggestion
If you're using any environment variables or **.env** files, add them to Netlify. You can do this by going to the **Site settings** and adding them to the **Build & deploy** section.
```

File: docs/pages/router/reference/api-routes.mdx (hunk 4 above)
Comment (EvanBacon) [reply docs/pages/router/reference/api-routes.mdx:345]: Those steps will need to be updated in the future, they should indicate that you use `--dir` with server mode.

**Review by amandeepmittal — APPROVED** [review]

**Pull request merged**